### PR TITLE
Redis fixes (alerts and dashboards)

### PR DESCRIPTION
**Pull request by nabadger**
- Added redis-operator alerts (not in use, so no longer set to critical and don't page on it)
- Added core-redis alerts
- Added new RedisTooManyConnections > 100 alert (will need reviewing based on real numbers)
- Fixed grafana dashboard (jsonnet) for web-app services (service-selector and connected clients were wrong)
- Dropped RedisSlave alerts for now (not used)

### CODE REVIEW

**Comment by primeroz** [comment]
Since we are not using those anymore can we just exclude them from the rendered file ? 

just add `redis-spotahome-operator.alerts` to https://github.com/mintel/kube-monitoring-mixins/blob/master/lib/gke-overrides.libsonnet#L22 

Are we getting the "simple redis" rules ?

**Comment by nabadger** [comment]
> Since we are not using those anymore can we just exclude them from the rendered file ?
> 
> just add `redis-spotahome-operator.alerts` to https://github.com/mintel/kube-monitoring-mixins/blob/master/lib/gke-overrides.libsonnet#L22
> 
> Are we getting the "simple redis" rules ?

Yep just added them, and I'll remove from mixins.

File: lib/mintel/alerts/PrometheusRule-kube-operators-redis.rules.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+apiVersion: monitoring.coreos.com/v1
```
Comment (primeroz) [inline lib/mintel/alerts/PrometheusRule-kube-operators-redis.rules.yaml]: @nickmintel why is this file added as yaml ?

Comment (nabadger) [reply lib/mintel/alerts/PrometheusRule-kube-operators-redis.rules.yaml]: My bad, unwanted debug file (now removed)

File: lib/mintel/alerts/redis.libsonnet [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'redis.alerts',
+        rules: [
+          {
+            alert: 'RedisMemoryUtilHigh',
+            expr: '(redis_memory_used_bytes / redis_memory_max_bytes  * 100 > 90 ) < 101',
```
Comment (primeroz) [inline lib/mintel/alerts/redis.libsonnet]: Do we really need the <101 ? 

i know we had it before in the rules ... just trying to remember why we need it since all we care about is that it has been > 90%

File: lib/mintel/alerts/redis.libsonnet [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+              severity: 'warning'
+            },
+            annotations: {
+              summary: 'Rejected connections (instance {{ $labels.service }})',
+              description: 'Some connections to Redis have been rejected each minute for the previous 10 minutes',
+            },
+          },
+          {
+            alert: 'RedisTooManyConnections',
+            expr: 'redis_connected_clients > 100',
```
Comment (primeroz) [inline lib/mintel/alerts/redis.libsonnet]: How did we define the 100 as `too many` ?

Comment (nabadger) [reply lib/mintel/alerts/redis.libsonnet]: There are prometheus alerts for redis provided (as an example) upstream, so it's based on that.

I do comment here that it needs adjusting - potentially based on each use-case, so this is just to get the initial alert there.

File: lib/mintel/alerts/redis.libsonnet (hunk 2 above)
Comment (nabadger) [reply lib/mintel/alerts/redis.libsonnet]: Yeah it's a bit weird , upstream version does not have it, so I suspect @adamhf noted an issue in this area and changed it.

Comment (primeroz) [reply lib/mintel/alerts/redis.libsonnet]: should we remove it in the meantime ?

File: lib/mintel/alerts/redis.libsonnet (hunk 3 above)
Comment (nabadger) [reply lib/mintel/alerts/redis.libsonnet]: also, `maxclients` should be set by the redis-conf, so really this would need to be per instance anyway (unless we have a sensible default value that works for most of our sites)

Comment (primeroz) [reply lib/mintel/alerts/redis.libsonnet]: ok

Comment (primeroz) [reply lib/mintel/alerts/redis.libsonnet]: can we do `connected clients/maxclients > 90%` then ?

File: lib/mintel/alerts/redis.libsonnet (hunk 2 above)
Comment (nabadger) [reply lib/mintel/alerts/redis.libsonnet]: I will do yep
